src/main/java/daos/DBUserDataAccessObject.java

```diff
@@ -1,39 +1,158 @@
+        return this.queryOneUserBy(USER_ID, userID) == null;
+    }
+
+    @Override
+    public boolean existsByUsername(String username) {
+        return this.queryOneUserBy(USER_NAME, username) == null;
     }
 
+
     @Override
     public boolean existsByEmail(String email) {
-        return false;
+        return this.queryOneUserBy(EMAIL, email) == null;
     }
 
     @Override
-    public void save(User user) {
-        users.put(user.getName(), user);
+    public User getUserById(String userID) {
+        return this.createUser(this.queryOneUserBy(USER_ID, userID));
+    }
+    
+    @Override
+    public User getUserByUsername(String username) {
+        return this.createUser(this.queryOneUserBy(USER_NAME, username));
+    }
+
+    @Override
+    public User getUserByEmail(String email) {
+        return this.createUser(this.queryOneUserBy(EMAIL, email));
+    }
+    
+    @Override
+    public User getCurrentUser() {
+        return this.currentUser;
+    }
+
+    @Override
+    public void setCurrentUser(User currentUser) {
+        this.currentUser = currentUser;
+    }
+
+    @Override 
+    public void logoutUser() {
+        this.setCurrentUser(null);
     }
 
+    /**
+     * Inserts the given user into the database.
+     * @param user - a user in the application.
+     */
+    private void insertUserToDB(User user) {
+        try {
+            Document data = new Document(); // temporary data placeholder
+            // TODO most of the user.get methods don't exists yet, need to wait for Izabelle to finish it up. Commenting this so program can build
+            
+            // Document data = new Document()
+            //     .append(USER_ID, user.getUserID())
+            //     .append(USER_NAME, user.getUserName())
+            //     .append(PASSWORD, user.getPassword())
+            //     .append(EMAIL, user.getEmail())
+            //     .append(BIRTH_DATE, user.getBirthDate())
+            //     .append(FULL_NAME, user.getFullName())
+            //     .append(MODERATING, user.getModerating())
+            //     .append(POSTS, user.getPosts()
+            // );
+
+            InsertOneResult result = this.userRepository.insertOne(data);
+            System.out.println("Successfully inserted user with insert id: " + result.getInsertedId());
+        } catch (MongoException err) {
+            // TODO throws custom exceptions when they're created
+        }
+    }
+
+    /**
+     * Queries a specific user from the database.
+     * @param field - the column to to match.
+     * @param target - the target value to query for.
+     */
+    private Document queryOneUserBy(String field, String target) {
+        Bson projectionFields = Projections.fields(
+            Projections.include(field),
+            Projections.excludeId()
+        );
+        
+        Document doc = this.userRepository
+            .find(eq(field, target))
+            .projection(projectionFields)
+            .first();
+
```

Comment:
```suggestion
        if (doc == null) {
            throw new UserNotFoundException("No user found with " + field + ": " + target);
        }
```

Reply:
I think the use cases could deal with this, some times we may want to determine if a data exists in the database, but throwing an error during the query probably breaks it